*** ../orig/zlib-1.1.4/configure	Wed Jul  8 14:19:35 1998
--- configure	Sun Feb  9 11:11:19 2003
***************
*** 18,23 ****
--- 18,24 ----
  # If you have problems, try without defining CC and CFLAGS before reporting
  # an error.
  
+ LDFLAGS="-L. -lz"
  LIBS=libz.a
  SHAREDLIB=libz.so
  VER=`sed -n -e '/VERSION "/s/.*"\(.*\)".*/\1/p' < zlib.h`
***************
*** 116,121 ****
--- 117,128 ----
  	     SFLAGS=${CFLAGS-"-Kconform_pic -O"}
  	     CFLAGS=${CFLAGS-"-O"}
  	     LDSHARED=${LDSHARED-"cc -G"};;
+   AIX*)     
+ 	     SFLAGS=${CFLAGS-"-O -qmaxmem=8192"}
+ 	     CFLAGS=${CFLAGS-"-O -qmaxmem=8192"}
+ 	     LDTESTSHARED=${LDSHARED-"cc -G"}
+ 	     LDSHAREDFLAGS="-L. libz.so"
+ 	     LDSHARED=${LDSHARED-"cc -G"};;
    # send working options for other systems to support@gzip.org
    *)         SFLAGS=${CFLAGS-"-O"}
  	     CFLAGS=${CFLAGS-"-O"}
***************
*** 127,135 ****
    echo Checking for shared library support...
    # we must test in two steps (cc then ld), required at least on SunOS 4.x
    if test "`($CC -c $SFLAGS $test.c) 2>&1`" = "" &&
!      test "`($LDSHARED -o $test$shared_ext $test.o) 2>&1`" = ""; then
      CFLAGS="$SFLAGS"
      LIBS="$SHAREDLIB.$VER"
      echo Building shared library $SHAREDLIB.$VER with $CC.
    elif test -z "$old_cc" -a -z "$old_cflags"; then
      echo No shared library suppport.
--- 134,143 ----
    echo Checking for shared library support...
    # we must test in two steps (cc then ld), required at least on SunOS 4.x
    if test "`($CC -c $SFLAGS $test.c) 2>&1`" = "" &&
!      test "`($LDTESTSHARED -o $test$shared_ext $test.o) 2>&1`" = ""; then
      CFLAGS="$SFLAGS"
      LIBS="$SHAREDLIB.$VER"
+ 	LDFLAGS="$LDSHAREDFLAGS"
      echo Building shared library $SHAREDLIB.$VER with $CC.
    elif test -z "$old_cc" -a -z "$old_cflags"; then
      echo No shared library suppport.
***************
*** 209,212 ****
--- 217,221 ----
  /^exec_prefix *=/s%=.*%=$exec_prefix%
  /^libdir *=/s%=.*%=$libdir%
  /^includedir *=/s%=.*%=$includedir%
+ /^LDFLAGS *=/s%=.*%=$LDFLAGS%
  " > Makefile
